src/libcore/iter.rs

```diff
@@ -1791,6 +1794,9 @@ impl<T, I> Iterator for Peekable<T, I> where I: Iterator<Item=T> {
 }
 
 #[stable]
+impl<T, I> ExactSizeIterator for Peekable<T, I> where I: ExactSizeIterator<Item = T> {}
```

Comment:
(copying over from previous comment to continue discussion)

> Sorry, that was supposed to be (uint::MAX, Some(uint::MAX)) (copy and paste error).
> 
> My basic point is that that if the number of items in an ExactSizeIterator once fit in a uint (required by the spec), it must always fit in a uint regardless of where those items are (moving one from self.iter to self.peeked doesn't change the total number of items).

Ah right, good point! So if the underlying iterator returns `(MAX, Some(MAX))` then the only time we'll add 1 is if we've already peeked in which case the underlying iterator will return `(MAX-1, Some(MAX-1))`.